google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeployAction.java

```diff
@@ -247,15 +241,72 @@ public void run(@NotNull ProgressIndicator indicator) {
 
   }
 
-  private String parseDeployOutputToModule(String jsonOutput)
+  // Just to return two strings.
+  static private class VersionService {
+    public String version = null;
+    public String service = null;
+  }
+
+  private VersionService parseDeployOutputToModule(String jsonOutput)
       throws JsonParseException {
-    Type deployOutputType = new TypeToken<Map<String, String>>() {}.getType();
-    Map<String, String> deployOutput = new Gson().fromJson(jsonOutput, deployOutputType);
+    VersionService versionService = new VersionService();
+
+    /* An example JSON output of gcloud app deloy:
+        {
+          "configs": [],
+          "versions": [
+            {
+              "id": "20160429t112518",
+              "last_deployed_time": null,
+              "project": "springboot-maven-project",
+              "service": "default",
+              "traffic_split": null,
+              "version": null
+            }
+          ]
+        }
+    */
 
-    if(deployOutput == null || deployOutput.keySet().size() != 1) {
-      throw new AssertionError("Expected a single module output from flex deployment.");
+    JsonReader reader = new JsonReader(new StringReader(jsonOutput));
+    try {
+      reader.beginObject(); // Top-level is a single object.
+
+      // Look for "versions" in the top-level object.
+      while (reader.hasNext()) {
```

Comment:
That would be nice, actually. My only concern is the possibility of future changes to the JSON output. Will that kind of marshallable object resilient/flexible enough to adapt changes in JSON. I hadn't tested it, but it feels like if anything is out of the type slightly, Gson will fail.

Related to that, it is not clear, for example, if `"traffic_split": null` will have a string type or may have an array.